latch_cli/centromere/ctx.py

```diff
@@ -223,28 +238,98 @@ def full_image(self):
-            headers=headers,
-            json={
-                "public_key": self.public_key,
-            },
+        json_data = resp.json()
+        if resp.status_code != 200:
+            raise ValueError(json_data["Error"])
+
+        ip = json_data["IP"]
+        self.jump_key_path.write_text(json_data["JumpKey"])
+        self.jump_key_path.chmod(0o600)
+
+        subprocess.run(
+            ["ssh-add", str(self.jump_key_path)],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
         )
 
-        resp = response.json()
-        try:
-            public_ip = resp["ip"]
-            username = resp["username"]
-        except KeyError as e:
+        retries = 0
+        while retries < 120:
+            resp = tinyrequests.post(
+                "https://centromere.latch.bio/register/ready",
+                headers={"Authorization": f"Latch-SDK-Token {self.token}"},
+            )
+
+            if resp.status_code != 200:
+                raise ValueError(resp.json()["Error"])
+
+            if resp.json()["Ready"]:
+                break
+
+            retries += 1
+            time.sleep(1)
+
+        if retries == 120 or ip == "":
             raise ValueError(
-                f"Malformed response from request for access token {resp}"
-            ) from e
-        return public_ip, username
+                "Unable to provision registration server due to load - please try again"
+                " later."
+            )
+
+        return ip, "root"
+
+    def downscale_register_deployment(self):
+        resp = tinyrequests.post(
+            "https://centromere.latch.bio/register/stop",
+            headers={"Authorization": f"Latch-SDK-Token {self.token}"},
+        )
+
+        if resp.status_code != 200:
+            raise ValueError("unable to downscale register deployment")
+
+    def configure_ssh_config(
+        self,
+        internal_ip: str,
+    ):
+        self.ssh_config_path = Path.home() / ".ssh" / "config"
+
+        self.ssh_config_path.parent.mkdir(exist_ok=True)
+        self.ssh_config_path.touch(exist_ok=True)
+
+        self.old_ssh_config = self.ssh_config_path.read_text()
+
+        jump_host_config = dedent(f"""
+            # >>> LATCH
+
+            Host latch_register
+                HostName {internal_ip}
+                ProxyJump {latch_constants.jump_user}@{latch_constants.jump_host}
+
+            # LATCH <<<
+            """).strip("\n")
+
+        match = ssh_config_expr.search(self.old_ssh_config)
+
+        if match:
+            ssh_config = ssh_config_expr.sub(jump_host_config, self.old_ssh_config)
+        else:
+            ssh_config = "\n".join([self.old_ssh_config, jump_host_config])
+
+        self.ssh_config_path.write_text(ssh_config)
+
+    def restore_ssh_config(self):
+        if self.old_ssh_config is None or self.ssh_config_path is None:
+            return
+
+        self.ssh_config_path.write_text(self.old_ssh_config)
```

Comment:
you should replace the regex with empty string instead, in case somebody modified the file while this code was running

also ideally we lock the config file (`fadvise`) when we open it because otherwise there is a race here since our modification is not atomic

Reply:
added a todo for the fadvise